File: src/test/cpp/model/castlingtypetest.cpp
/*
 * Copyright (C) 2013-2019 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

#include "model/castlingtype.h"

#include "gtest/gtest.h"

using namespace pulse;

TEST(castlingtypetest, testValues) {
	for (auto castlingtype: castlingtype::values) {
		EXPECT_EQ(castlingtype, castlingtype::values[castlingtype]);
	}
}

File: pulse-java/src/test/java/com/fluxchess/pulse/model/CastlingTypeTest.java
/*
 * Copyright 2013-2023 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse.model;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class CastlingTypeTest {

	@Test
	void testValues() {
		for (int castlingtype : CastlingType.values) {
			assertThat(CastlingType.values[castlingtype]).isEqualTo(castlingtype);
		}
	}
}
